### PR TITLE
Ambient Light

### PR DESCRIPTION
**Pull request by Lohikar**
Adds basic support for ambient lighting -- effectively radius 2 lights that can only be emitted by turfs.
These lights are much cheaper than using normal lights, at the cost of being much more limited.

Todo:
- [x] VV hook
- [x] Improve naming
- [x] Add setter (ala `set_light()`)

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
````
[4:27 PM] MistakeNot: maybe rename og ambient light proc to refresh
[4:27 PM] MistakeNot: or rename yours to update_ambient_light_corners or smt
[4:27 PM] Lohikar: prob better to leave that as a comment on the PR
````

**Comment by MistakeNot4892** [comment]
Noting that this obviates/properly implements what was planned by #2174

**Comment by Lohikar** [comment]
This is 'done', but it still has some odd exoplanet bugs left.

File: code/game/turfs/exterior/_exterior.dm [hunk 1]
```diff
@@ -25,11 +25,11 @@
 		owner = null
 	else
 		//Must be done here, as light data is not fully carried over by ChangeTurf (but overlays are).
```
Comment (MistakeNot4892) [inline code/game/turfs/exterior/_exterior.dm:27]: Might need to remove or move this comment?

**Comment by MistakeNot4892** [comment]
Pulled this into https://github.com/NebulaSS13/Nebula/pull/2432 to test it post-overmap adjustments, and ~~the lighting artifacts seem to be gone after moving the exoplanet gen down behind SSatoms~~ the lighting artifacts persist.

However it seems like ChangeTurf is not properly clearing ambient light:
https://cdn.discordapp.com/attachments/678826323728400394/981104117395112026/unknown.png

These turfs are `ambient_light = null` but the corners are all 0.3

**Comment by MistakeNot4892** [comment]
Interesting artifact: 
image

**Comment by MistakeNot4892** [comment]
Alrighty yeah artifacts are still happening. 
image

**Comment by MistakeNot4892** [comment]
image
Drop Bomb on exterior walls has some interesting effects.

**Comment by MistakeNot4892** [comment]
Day-night transition is very messed up: 
image

**Comment by MistakeNot4892** [comment]
Setting light on the middle turf via proccalled set_ambient_light(#ffffff, 0.5): 
image

**Comment by MistakeNot4892** [comment]
Artifacts are much improved, but still seeing some weirdness: https://cdn.discordapp.com/attachments/678826323728400394/982169459013853184/unknown.png

**Comment by MistakeNot4892** [comment]
This has some mild artifacts remaining but so does the current lighting engine due to exoplanet turf init being very strange. Happy to merge this as is and resolve the artifacts over time.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**